parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.parser.sqllogictest.java;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTest.AcceptedRdbms.ALL;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestConfig.TEST_FILES_ROOT_PATH;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.createFileForResults;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.validateStatementAgainstPossibleErrors;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.writeTestResultToFile;
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static java.nio.file.Files.newBufferedReader;
+import static java.nio.file.Files.walkFileTree;
+
+public final class SqlLogicTest
+{
+    private SqlLogicTestConfig config;
+    private SqlLogicTestResult result;
+
+    private Set<AcceptedRdbms> skipIfSet;
+    private AcceptedRdbms onlyIf;
+    private FileVisitor<Path> fileVisitor;
+
+    protected enum SqlLogicTestError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ORIGINAL_ERROR
+    }
+
+    protected enum AcceptedRdbms
```

Comment:
Feels like this class should be named `Rdbms` rather than `AcceptedRdbms`. I might just name it `Database` or `DbDialect` in this context because we really only mean their SQL dialect, not the relational database management system.

Reply:
I agree -- dialect is more accurate. Switching this enum name to DatabaseDialect.

Done.